internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -468,6 +548,26 @@ func (r *AgentServiceConfigReconciler) newPostgresSecret(ctx context.Context, lo
 	return secret, mutateFn, nil
 }
 
+func (r *AgentServiceConfigReconciler) newImageHandlerServiceAccount(ctx context.Context, log logrus.FieldLogger, instance *aiv1beta1.AgentServiceConfig) (client.Object, controllerutil.MutateFn, error) {
```

Comment:
The reason I created a new service account is because the assisted-service has the ability to watch and manipulate things at the cluster scope and the image-service shouldn't need to talk to the api at all.

Reply:
Right, forgot about the controller permissions. :man_facepalming: